sktime/utils/seasonality.py

```diff
@@ -193,24 +201,36 @@ def _unpivot_sp(df, template=None):
             freq = template.freq
         else:
             freq = template
-
+    else:
+        freq = None
+    if isinstance(df.index, pd.DatetimeIndex):
```

Comment:
is handling `tz` separate issue?

Reply:
Not sure, this issue was triggered when trying to make the test green.

Reply:
how does `tz` come into everything?

Reply:
I am not sure if I understand this correctly. `tz` is only used if the time series has a datetimeindex.

Reply:
yes, but what I mean, why did you feel you had to add logic related to `tz`, where previously there was none.

Is it dependent, i.e., you tried to fix, and then `tz` related problems popped up, or is it independent, i.e., there is another issue related to `tz` that you also decided o fix.

Reply:
It popped up when trying to fix the period issues.

Reply:
typical `pandas` trouble... 😢